Src/ToastNotifications/Display/NotificationsDisplaySupervisor.cs

```diff
@@ -41,7 +41,10 @@ public class NotificationsDisplaySupervisor : IDisposable
 
         public void DisplayNotification(INotification notification)
         {
-            Dispatch(() => InternalDisplayNotification(notification));
+            if (_dispatcher.CheckAccess())
+                InternalDisplayNotification(notification);

```

Comment:
This is the fix:  Dispatch() would run the message pump resulting in a recursive call.